website/content/docs/job-specification/vault.mdx

```diff
@@ -58,6 +58,17 @@ with Vault as well.
 
 ## `vault` Parameters
 
+- `allow_token_expiration` `(bool: false)` - Specifies that Nomad clients should
+  not attempt to renew a task's Vault token, allowing it to expire. This should
+  only be used when a secret is requested from Vault once at the start of a task
+  or in a short-lived prestart task. Long-running tasks should never set
+  `allow_token_expiration=true` if they obtain Vault secrets via `template`
+  blocks, as the Vault token will expire and the template runner will continue
+  to make failing requests to Vault until its [`vault_retry`][] attempts are
+  exhausted, at which point the task will fail. This field is valid only when
+  Nomad has been configured to use [Workload Identity with Vault][] and will be
+  ignored when using the [legacy Vault authentication workflow][].
```

Comment:
It's not fully ignored. I think the token property is ignored, but the config still works even with the legacy workflow.

It could also be useful to mention that, if the Vault token is already configured as non-reneweable then you don't need to set this.

Reply:
Done in 5798001 along with your suggestion to add a note to the Vault ACL docs. How's that look to you?

Reply:
Expanded on that some more in 183d356